packages/custom-storage/__tests__/SQLStorage.spec.ts

```diff
@@ -0,0 +1,179 @@
+import { SQL } from '@carto/toolkit-sql';
+import { SQLStorage } from '../src/sql/SQLStorage';
+import {
+  StoredVisualization,
+  // Visualization,
+  // Dataset
+} from '../src/StorageRepository';
+import {
+  CHECK_CREATE_QUERIES,
+  LIST_VISUALIZATION_QUERY
+} from './mocks/MockedQueries';
+
+describe('SQLStorage', () => {
+  let sqlStorage: SQLStorage;
+
+  beforeAll(() => {
+    const sqlClient = new SQL('user', 'm14p1k3y', 'https://{user}.carto.com/');
+    sqlStorage = new SQLStorage('mynamespace', sqlClient, 1, true);
+  });
+
+  it('should check or create tables for storage', async () => {
+    // We use only one mock because is not important for the rest of the queries
+    (global as any).fetch.mockResponse(
+      JSON.stringify({
+        rows: [
+          {rolename: 'cartodb_publicuser_a1b2c3d4f5'}
+        ],
+        time: 0.001,
+        fields: {
+          rolename: {type: 'name', pgtype: 'name'}
+        },
+        total_rows: 1
+      }),
+      {
+        headers: [
+          ['content-type', 'application/json'],
+          ['Retry-After', 0],
+          ['Carto-Rate-Limit-Remaining', 5]
+        ]
+      }
+    );
+    await sqlStorage.init();
+
+    // Check the creation and access grant queries of the 3 tables
+    (global as any).fetch.mock.calls.forEach((call: any, idx: number) => {
+      const params = (global as any).getURLParams(call[0]);
+      expect(decodeURI(params.q)).toBe(CHECK_CREATE_QUERIES[idx]);
+    });
+  });
+
+  it('should list visualizations', async () => {
+    const visObject: StoredVisualization = {
```

Comment:
Maybe we can extract the content into a DATA const, to reuse both here and in the mock

Reply:
:+1: